src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1457,7 +1457,7 @@ private MethodSymbol FindEntryPoint(CancellationToken cancellationToken, out Imm
                 var viableEntryPoints = ArrayBuilder<MethodSymbol>.GetInstance();
                 foreach (var candidate in entryPointCandidates)
                 {
-                    if (!candidate.HasEntryPointSignature(this))
+                    if (!candidate.HasEntryPointSignature(this, diagnostics))
```

Comment:
>diagnostics [](start = 64, length = 11)

As we discussed offline, I believe we should pass ```warnings``` here. The fact that you didn't catch that in your tests indicates that you probably do not have any test where there are more than one Main candidate. One has errors, but the other one is good (a regular [void Main()] for example). Please add a test like this.